polygeist/lib/Dialect/Polygeist/Transforms/BareMemRefToLLVM.cpp

```diff
@@ -319,6 +319,21 @@ struct StoreMemRefOpLowering : public MemAccessLowering {
     return success();
   }
 };
+
+struct ViewMemRefOpLowering : public ConvertOpToLLVMPattern<memref::ViewOp> {
+  using ConvertOpToLLVMPattern<memref::ViewOp>::ConvertOpToLLVMPattern;
+
+  LogicalResult
+  matchAndRewrite(memref::ViewOp viewOp, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    const auto convElemType = typeConverter->convertType(
+        viewOp.getSource().getType().getElementType());
```

Comment:
```suggestion
    if (!canBeLoweredToBarePtr(viewOp.getSource().getType()) || 
        !canBeLoweredToBarePtr(viewOp.getType()))
      return failure();
    const auto convElemType = typeConverter->convertType(
        viewOp.getSource().getType().getElementType());
    if (!convElemType)
      return viewOp.emitWarning("Target descriptor type not converted to LLVM");
```

Reply:
Why do we only check if conversion works for `memref::ViewOp`?

Reply:
I saw some other patterns not using `canBeLoweredToBarePtr`, this can lead to potential issues, so we should check that always.

Reply:
I mean 
```
    if (!convElemType)
      return viewOp.emitWarning("Target descriptor type not converted to LLVM");
```
I don't see other patterns doing that in the file.

Reply:
I'd say it's still a good practice to do so. Otherwise, we will run into errors if the type is invalid (`nullptr`).

Reply:
done.